Improve KeepAlive agent for node4

Raynos opened the pull request:
This change makes the KeepAlive agent
compatible with both node0.10 & node4

Making the HTTPS KeepAlive agent compatible is non-trivial
so I've opted to just delete it.

r: @rf @Matt-Esch

rf commented on test/endpoint_test.js:
```diff
@@ -33,7 +33,7 @@ describe("PoolEndpoint", function () {
     });
 
-    it("passes agentOptions to the underlying Agent (keep-alive secure)", function () {
+    it.skip("passes agentOptions to the underlying Agent (keep-alive secure)", function () {
```
why skip?

Raynos replied:
The test fails because no HTTPS :( I can delete it instead probably...

Matt-Esch commented:
Will need to go into a new major. Losing https is a significant feature drop. I agree that terminating https somewhere else is the recommended approach, but that doesn't offer a solution to consuming from https services. Personally I'm not all that convinced that removing it is the right thing if it still works in 0.10

Raynos commented:
I can undelete the code and tests.

I can just skip the test.

Raynos commented:
Addressed feedback, cc @Matt-Esch , simply skipping test in node4 and leaving HTTPsAgent unworking.

Because of the node4 implementation, it already throws an uncaught on construction so its a fail fast footgun.

Matt-Esch commented:
👍 thanks for addressing the comments
